### ISSUE TITLE
GUI not copied once plugin is installed

### ISSUE THREAD

**Issue by gmanni66** [brummer10/GxPlugins.lv2#37]
Hello brummer10
not sure how to contact you but first I want to thank you for the plugins
I wanted to report though that I've tried a couple of times to build these and I found the same problem both times
all the plugins build and work fine, however, the MOD/* files do not get copied in the appropriate folder 
I couldn't find why that is (yet) but wanted to report and maybe you know where I'm going wrong here 

is this normal? or there is a mistake in the install script?

let me know also if there is a better way to contact you rather than opening an issue

however, wanted to report this - thanx
PS: copying manually the files works just fine

**Comment by brummer10** [comment]
Yes, the makefiles been made to build the generic GUI's or the MOD GUI's, not both. At the time I wrote them there simply wasn't a host which may use them on PC.

**Comment by gmanni66** [comment]
Thanx for the reply,
Any ways to change/customize that??

**Comment by brummer10** [comment]
Main issue is that I've chosen to use different *.ttl files for the generic and the MOD GUI's. They need to be rewritten so that both GUI's could use the same *.ttl files. That wouldn't be to hard to do, but as there are a lot plug's in the package, it will require a lot time. When that is done, we could edit the makefiles to allow the installation of both GUI's at once.

**Comment by gmanni66** [comment]
Thank you for the reply
I guess I can just run a script to copy the MOD GUI into the respective folder before running the make install :)
If I come up with an easy script I'll let you know
Thanx

**Comment by gmanni66** [comment]
ok this simple script should do it 
you need to have it in the folder **GxPlugins.lv2** and run it before the "make install" so that all the modgui will be included 
NOTE: this is only for the modgui 
1 create the script
2 run it after the make (but before the make install) 
3 run make install 
script example

`#!/bin/bash

mkdir tmp
currentdir=`pwd`
echo "Current Directory :" $currentdir

for i in `ls -1 -d $PWD/*` ; do cd $i 2> /dev/null; ls -1 -d $PWD/* | grep -i "gx_" ; done > tmp/gxlist1
cd $currentdir
cat tmp/gxlist1 | sort | uniq > tmp/gxlist_clean

for i in `cat tmp/gxlist_clean` ; do cd $i ; mv ../MOD/* $i ; done

cd $currentdir
rm -rf tmp
`
hope this helps someone with the same issue I found :)

**Comment by Boone777** [comment]
Where do you manually copy the mod folder ?
I tried replacing everything in the .LV2 or just adding no duplicate from my make folder that did not work I only have generic GUIs.
Can somebody explain for a noob ?


I get all kinds of error trying the script but I'm a noob so does not mean much.

**Comment by brummer10** [comment]
To get both, the generic and the MOD UI you must run make first. Then you must go into the single plugin folders. For example ~/GxPlugins.lv2/GxAxisFace.lv2. There you find a folder called gx_AxisFace.lv2 That is the bundle to be installed. Now, to get the MOD UI into it go back to GxAxisFace.lv2, there you find a folder MOD This contain the MOD UI. Open that and copy modgui.ttl mainfest.ttl and the folder modgui. Don't copy the file gx_AxisFace.ttl, this is a modified file and allow only the MOD UI, so leave it there. Copy the rest over to gx_AxisFace.lv2 You'll get a warning because overwrite the file manifest.ttl Choose overwrite. 
After that run make install and you'll have both UI's installed. True, I don't know what you'll do with the MOD UI's on a desktop, is there a Host supporting MOD UI's? 
I keep this stuff separated because I use them on the MOD Dwarf, there you can't and wouldn't build the generic UI for.

**Comment by Boone777** [comment]
I actually thought MOD UI'S meant modified UI's, like a new skin. So these
can't be seen on AVLinux is that what I'm understanding here ?

**Comment by brummer10** [comment]
MOD UI's been web based UI's (html + css) to be used in the browser. As far I know only MOD could use them 
https://mod.audio/platform/
So, yes, you couldn't use them on AVLinux. 
Anyway, for the GxPlugins those didn't differ that much (from the view).

**Comment by Boone777** [comment]
I only get the generic Reaper UI out of the original install in AVLinux  something I should know about making the GUI show ?

**Comment by brummer10** [comment]
Ah, yes, I remember. There is a feature request in the ttl files which is deprecated. That needs to be removed. Sorry, I always come over it (It's only Reaper which fail on that). That's the line 
`lv2:requiredFeature guiext:makeResident;`

when you remove that line from the plug_name.ttl files you'll see the UI's. 
I'll try to update the sources as soon as possible.

**Comment by brummer10** [comment]
I've pushed a update now, please do a git pull and build again, should work now in reader as well.

**Comment by Boone777** [comment]
Thank you for working on this.
I'm not savvy at all when it comes to building it's hit and miss for me but I was able to do it with the one previously downloaded.
Here is what I am getting now:

```
mx21@mx:~/Downloads/GxPlugins.lv2-master
$ make
make[1]: Entering directory '/home/mx21/Downloads/GxPlugins.lv2-master/GxAxisFace.lv2'
make[1]: *** No targets specified and no makefile found.  Stop.
make[1]: Leaving directory '/home/mx21/Downloads/GxPlugins.lv2-master/GxAxisFace.lv2'
make: *** [Makefile:10: GxAxisFace.lv2/] Error 2

```

**Comment by Boone777** [comment]
> Ah, yes, I remember. There is a feature request in the ttl files which is deprecated. That needs to be removed. Sorry, I always come over it (It's only Reaper which fail on that). That's the line `lv2:requiredFeature guiext:makeResident;`
> 
> when you remove that line from the plug_name.ttl files you'll see the UI's. I'll try to update the sources as soon as possible.

This line for me is the very last. I removed it, saved and restart Reaper, clear cache rescan VST,  rescan LV2 still no GUI.

**Comment by brummer10** [comment]
I've done extensive tests in Reaper, it should work.
To download, build and install the last version, do 
```
git clone https://github.com/brummer10/GxPlugins.lv2.git
cd GxPlugins.lv2
git submodule init
git submodule update
make
make install
```

**Comment by Boone777** [comment]
Thank you !!!!! Everything works now I really appreciate your efforts !

**Comment by Boone777** [comment]
I actually have only one that is not showing GUI: Axisface

**Comment by brummer10** [comment]
Ah, Yea I've missed to push the update for the Axisface. Done now, please do a git pull and rebuild.
